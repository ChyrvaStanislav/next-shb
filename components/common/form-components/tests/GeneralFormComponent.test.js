import React, { useState } from 'react';
import { render, fireEvent } from '@testing-library/react';
import GeneralFormComponent from '../index';

// eslint-disable-next-line react/prop-types
const renderMock = ({ setFormSent }) => (
  <>
    <h1>Some children</h1>
    <button type="button" onClick={() => setFormSent(true)}>setFormSent</button>
  </>
);


const onDisableBody = jest.fn();
const onEnableBody = jest.fn();
const disableBody = () => {
  onDisableBody();
};
const enableBody = () => {
  onEnableBody();
};

jest.mock('../modal/utils', () => ({
  disableBody, enableBody,
}));

const Component = (props) => {
  const [open, setOpen] = useState(true);

  return (
    <GeneralFormComponent
      open={open}
      setOpen={setOpen}
      title="Some Title"
      render={renderMock}
      successScreenConfig={{
        formType: 'main-request',
        title: 'SuccessScreen Title',
        subDescriptionFirst: 'subDescriptionFirst',
        subDescriptionSecond: 'subDescriptionSecond',
      }}
      successScreenRenderButton={null}
      successScreenClassGA="successScreenClassGA"
      {...props}
    />
  );
};

describe('GeneralFormComponent renders', () => {
  it('should GeneralFormComponent renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});


describe('setFormSent works', () => {
  it('should setFormSent works', () => {
    const { getByText } = render(<Component />);
    const button = getByText('setFormSent');
    fireEvent.click(button);

    expect(getByText('SuccessScreen Title')).toBeInTheDocument();
  });
});
