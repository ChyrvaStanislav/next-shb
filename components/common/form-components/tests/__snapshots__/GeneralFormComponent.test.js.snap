// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralFormComponent renders should GeneralFormComponent renders 1`] = `
<DocumentFragment>
  <div>
    <section
      class="componentContainer"
      data-testid="section"
    >
      <div
        class="slider open"
      >
        <h1>
          Some children
        </h1>
        <button
          type="button"
        >
          setFormSent
        </button>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
